File: output/csharp/QueryDsl/FullText/QueryString/QueryStringQuery.cs
using Nest.CommonAbstractions;
using Nest.Internal;
using Nest.QueryDsl;
using Nest.CommonOptions;
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;
namespace Nest.QueryDsl {

	public class QueryStringQuery  {
		
		[DataMember(Name="allow_leading_wildcard")]
		public bool AllowLeadingWildcard { get; set; }


		[DataMember(Name="analyzer")]
		public string Analyzer { get; set; }


		[DataMember(Name="analyze_wildcard")]
		public bool AnalyzeWildcard { get; set; }


		[DataMember(Name="auto_generate_synonyms_phrase_query")]
		public bool AutoGenerateSynonymsPhraseQuery { get; set; }


		[DataMember(Name="default_field")]
		public Field DefaultField { get; set; }


		[DataMember(Name="default_operator")]
		public Operator DefaultOperator { get; set; }


		[DataMember(Name="enable_position_increments")]
		public bool EnablePositionIncrements { get; set; }


		[DataMember(Name="escape")]
		public bool Escape { get; set; }


		[DataMember(Name="fields")]
		public List<Field> Fields { get; set; }


		[DataMember(Name="fuzziness")]
		public Fuzziness Fuzziness { get; set; }


		[DataMember(Name="fuzzy_max_expansions")]
		public int FuzzyMaxExpansions { get; set; }


		[DataMember(Name="fuzzy_prefix_length")]
		public int FuzzyPrefixLength { get; set; }


		[DataMember(Name="fuzzy_rewrite")]
		public MultiTermQueryRewrite FuzzyRewrite { get; set; }


		[DataMember(Name="fuzzy_transpositions")]
		public bool FuzzyTranspositions { get; set; }


		[DataMember(Name="lenient")]
		public bool Lenient { get; set; }


		[DataMember(Name="max_determinized_states")]
		public int MaxDeterminizedStates { get; set; }


		[DataMember(Name="minimum_should_match")]
		public MinimumShouldMatch MinimumShouldMatch { get; set; }


		[DataMember(Name="phrase_slop")]
		public double PhraseSlop { get; set; }


		[DataMember(Name="query")]
		public string Query { get; set; }


		[DataMember(Name="quote_analyzer")]
		public string QuoteAnalyzer { get; set; }


		[DataMember(Name="quote_field_suffix")]
		public string QuoteFieldSuffix { get; set; }


		[DataMember(Name="rewrite")]
		public MultiTermQueryRewrite Rewrite { get; set; }


		[DataMember(Name="tie_breaker")]
		public double TieBreaker { get; set; }


		[DataMember(Name="time_zone")]
		public string TimeZone { get; set; }


		[DataMember(Name="type")]
		public TextQueryType Type { get; set; }

	}
}

File: specification/__common/query_dsl/full_text/query_string/QueryStringQuery.ts
/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import {
  double,
  Field,
  Fields,
  Fuzziness,
  integer,
  MultiTermQueryRewrite,
} from "../../../common";
import { MinimumShouldMatch } from "../../../common_options/minimum_should_match/MinimumShouldMatch";
import { QueryBase } from "../../abstractions/query/Query";
import { Operator } from "../../Operator";
import { TextQueryType } from "../multi_match/TextQueryType";

export class QueryStringQuery extends QueryBase {
  allow_leading_wildcard?: boolean;
  analyzer?: string;
  analyze_wildcard?: boolean;
  auto_generate_synonyms_phrase_query?: boolean;
  default_field?: Field;
  default_operator?: Operator;
  enable_position_increments?: boolean;
  escape?: boolean;
  fields?: Fields;
  fuzziness?: Fuzziness;
  fuzzy_max_expansions?: integer;
  fuzzy_prefix_length?: integer;
  fuzzy_rewrite?: MultiTermQueryRewrite;
  fuzzy_transpositions?: boolean;
  lenient?: boolean;
  max_determinized_states?: integer;
  minimum_should_match?: MinimumShouldMatch;
  phrase_slop?: double;
  query?: string;
  quote_analyzer?: string;
  quote_field_suffix?: string;
  rewrite?: MultiTermQueryRewrite;
  tie_breaker?: double;
  time_zone?: string;
  type?: TextQueryType;
}
